### PR TITLE
add transaction-pessimistic.md

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

File: sql/transaction-pessimistic.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+```
+[pessimistic-txn]
+enable = true
+default = true
+```
+
+在配置了默认启用悲观事务的情况下，可以用以下两种方式使事务进入乐观事务模式：
+
+1. 执行 `BEGIN OPTIMISTIC;` 语句开启的事务，会进入乐观事务模式。
```
Comment (tiancaiamao) [inline sql/transaction-pessimistic.md]: 有这个么？

File: sql/transaction-pessimistic.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+可以通过写成注释的形式 `BEGIN /*!90000 OPTIMISTIC */;`来兼容 MySQL 语法。
+
+2. 通过执行 `set @@tidb_optimistic_txn = 1;`，使当前的 session 执行的事务进入乐观事务模式。
```
Comment (tiancaiamao) [inline sql/transaction-pessimistic.md]: 变量太多了吧... 不如做成 set @@tidb_pessimistic_txn = auto/on/off
auto 根据配置里面 default 来

File: sql/transaction-pessimistic.md (hunk 1 above)
Comment (coocood) [reply sql/transaction-pessimistic.md]: 后面可以加

**Comment by tiancaiamao** [comment]
LGTM

**Comment by morgo** [comment]
Can we move this under the URL structure: `dev/reference/transactions/transactions-pessimistic.md` ?

This will save having to add a redirect as we move it later.

**Comment by coocood** [comment]
@morgo 
The `dev/reference` seems not exists.

**Comment by morgo** [comment]
It is coming soon. See https://github.com/pingcap/docs-cn/pull/1351

cc @dcalvin

**Comment by coocood** [comment]
@morgo 
We can wait for that PR to merge.

**Comment by dcalvin** [comment]
> @morgo
> We can wait for that PR to merge.

Yes, it will be soon merged. Thanks @morgo  for the reference.

**Comment by dcalvin** [comment]
@coocood FYI #1351 has been merged

File: TOC.md [hunk 3]
```diff
@@ -124,6 +124,7 @@
   + 事务
     - [事务模型](sql/transaction-model.md)
     - [隔离级别](sql/transaction-isolation.md)
+    - [悲观事务](sql/transaction-pessimistic.md)
```
Comment (dcalvin) [inline TOC.md]: @coocood  Please pull the latest TOC from master and add this entry as below: 
```
+ 事务
    - [事务模型](dev/reference/transactions/transaction-model.md)
    - [隔离级别](dev/reference/transactions/transaction-isolation.md)
    - [悲观事务](dev/reference/transactions/transaction-pessimistic.md)
```
In this PR, you will also need to update the file location as indicated in TOC.

**Comment by coocood** [comment]
@dcalvin
thanks, I will resolve the conflict later.

**Comment by coocood** [comment]
@dcalvin 
PTAL

File: dev/reference/transactions/transaction-pessimistic.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+title: TiDB 悲观事务模式
+category: user guide
+---
+
+# TiDB 悲观事务模式
+
+TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败的问题，为了保证事务的成功率，需要修改应用程序，加上重试的逻辑。
+悲观事务模式可以避免这个问题，应用程序无需添加重试逻辑，就可以正常执行。
+
+悲观事务模式目前是一个试验性的特性，*不建议在生产环境中使用*。
```
Comment (dcalvin) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
> **注意：**
>
> 悲观事务模式目前是一个**试验性**的特性，*不建议在生产环境中使用*。
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
+---
+title: TiDB 悲观事务模式
+category: user guide
```
Comment (dcalvin) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
category: reference
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+---
+
+# TiDB 悲观事务模式
+
+TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败的问题，为了保证事务的成功率，需要修改应用程序，加上重试的逻辑。
```
Comment (dcalvin) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
TiDB 默认使用乐观事务模式，存在事务提交时因为冲突而失败的问题。为了保证事务的成功率，需要修改应用程序，加上重试的逻辑。
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+因为悲观事务目前还是一个试验性特性，默认是关闭的，启用悲观事务首先需要在 TiDB 的配置文件里添加:
+
+```
+[pessimistic-txn]
+enable = true
+```
+
+`enable` 设置为 `true` 以后，默认的事务模式仍然是乐观事务模式，要进入悲观事务模式有以下三种方式:
+
+1. 执行 `BEGIN PESSIMISTIC;` 语句开启的事务，会进入悲观事务模式。
```
Comment (dcalvin) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
- 执行 `BEGIN PESSIMISTIC;` 语句开启的事务，会进入悲观事务模式。
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+可以通过写成注释的形式 `BEGIN /*!90000 PESSIMISTIC */;`来兼容 MySQL 语法。
+
+2. 执行 `set @@tidb_txn_mode = 'pessimistic';`，使这个 session 执行的所有事务都会进入悲观事务模式。
```
Comment (dcalvin) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
- 执行 `set @@tidb_txn_mode = 'pessimistic';`，使这个 session 执行的所有事务都会进入悲观事务模式。
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+3. 在配置文件设置默认启用悲观事务模式，使除了自动提交的单语句事务之外的所有事务都会进入悲观事务模式。
```
Comment (dcalvin) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
- 在配置文件设置中默认启用悲观事务模式，使除了自动提交的单语句事务之外的所有事务都会进入悲观事务模式。
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+```
+[pessimistic-txn]
+enable = true
+default = true
+```
+
+在配置了默认启用悲观事务的情况下，可以用以下两种方式使事务进入乐观事务模式：
+
+1. 执行 `BEGIN OPTIMISTIC;` 语句开启的事务，会进入乐观事务模式。
```
Comment (dcalvin) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
- 执行 `BEGIN OPTIMISTIC;` 语句开启的事务，会进入乐观事务模式。
```
`BEGIN OPTIMISTIC;` 里的分号是多余的吗？

File: dev/reference/transactions/transaction-pessimistic.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+可以通过写成注释的形式 `BEGIN /*!90000 OPTIMISTIC */;`来兼容 MySQL 语法。
+
+2. 通过执行 `set @@tidb_txn_mode = 'optimistic';`，使当前的 session 执行的事务进入乐观事务模式。
```
Comment (dcalvin) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
- 通过执行 `set @@tidb_txn_mode = 'optimistic';`，使当前的 session 执行的事务进入乐观事务模式。
```
`set @@tidb_txn_mode = 'optimistic';`同上，分号多余？

File: dev/reference/transactions/transaction-pessimistic.md (hunk 7 above)
Comment (dcalvin) [reply dev/reference/transactions/transaction-pessimistic.md]: 此处不是具有顺序的操作步骤，需要使用无序列表 （bullet list）

File: dev/reference/transactions/transaction-pessimistic.md (hunk 10 above)
Comment (coocood) [reply dev/reference/transactions/transaction-pessimistic.md]: 不是多余的，其他文档里也是加了分号的。

**Comment by coocood** [comment]
@dcalvin PTAL

**Pull request merged**